### PR TITLE
Add Testing Library setup and some boilerplate tests for current comp…

### PR DESCRIPTION
**Pull request by adamjarling**
…onents

This more or less should wrap up the intent of #24 .  The tests are stubbed out mainly verifying the presence of layout elements.  Once I or anyone connects to a back end to see this running in action, it'll be easier to add more behavioral tests to each component.

Any questions on Testing Library, let me know.  Everything should follow the packages doc's fairly closely.

### CODE REVIEW

File: src/__tests__/ResourceTree.test.js [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+  const { getByTestId } = render(<ResourceTree {...defaultProps} />)
+  expect(getByTestId('primary-checkbox')).toBeInTheDocument()
+})
+
+it('renders expand/collapse button', () => {
+  const { getByTestId } = render(<ResourceTree {...defaultProps} />)
+  expect(getByTestId('expand-collapse-button')).toBeInTheDocument()
+})
+
+// Not really sure yet how to test <StyledChildResourceTree /> and
```
Comment (jrgriffiniii) [inline src/__tests__/ResourceTree.test.js]: 👍 I can create an issue for this.

File: src/__tests__/testing-helpers.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// you can provide initialState for the entire store that the ui is rendered with
+export function renderWithRedux(
+  ui,
+  { initialState, store = configureStore(initialState) } = {}
+) {
+  return {
+    ...render(<Provider store={store}>{ui}</Provider>),
+    // adding `store` to the returned utilities to allow us
+    // to reference it in our tests (just try to avoid using
+    // this to test implementation details).
```
Comment (jrgriffiniii) [inline src/__tests__/testing-helpers.js]: 👍

File: src/containers/SelectProvider.js [hunk 3]
```diff
@@ -1,57 +1,63 @@
...
-              return <MenuItem
-                       key={provider.id}
-                       value={provider.id}
-                     >{provider.name}</MenuItem>
-            })
-          }
-        </Select>
-      </FormControl>
-    );
+const useStyles = makeStyles({
```
Comment (jrgriffiniii) [inline src/containers/SelectProvider.js]: Thank you for adding this, I know that the usage of the Material UI should be adjusted, I can create an issue for this as well.

Comment (adamjarling) [reply src/containers/SelectProvider.js]: Sounds good.  We can only use the Material UI Hooks way of applying styles to functional components.  So in the more involved, class components I left the current Material UI style.

If you wanted to create an issue for converting class components to Hooks, I'd be happy to help out with that if you want?   And for that work, does a dev environment exist to run the front end working off the back-end to see it in action?

Comment (jrgriffiniii) [reply src/containers/SelectProvider.js]: I've proposed some documentation with https://github.com/samvera-labs/browse-everything-redux-react/pull/35, and I've updated https://github.com/samvera-labs/browse-everything-redux-react/issues/32 to be scoped for functional components with Hooks. Thank you very much for suggesting this!

**Pull request merged**